sdk/extensions/wcf/Microsoft.CoreWCF.Azure.StorageQueues/README.md

```diff
@@ -0,0 +1,152 @@
+{
+    app.UseServiceModel(serviceBuilder =>
+    {
+        // Configure CoreWCF to dispatch to service type Service
+        serviceBuilder.AddService<Service>();
+
+        // Create a binding instance to use Azure Queue Storage, passing an optional queue name for the dead letter queue
+        var aqsBinding = new AzureQueueStorageBinding("DEADLETTERQUEUENAME");
+
+        // Configure the client credential type to use StorageSharedKeyCredential
+        aqsBinding.Security.Transport.ClientCredentialType = AzureClientCredentialType.StorageSharedKey;
+
+        // Add a service endpoint using the AzureQueueStorageBinding
+        string queueEndpointString = "https://MYSTORAGEACCOUNT.queue.core.windows.net/QUEUENAME";
+        serviceBuilder.AddServiceEndpoint<Service, IService>(aqsBinding, queueEndpointString);
+
+        // Use extension method to configure CoreWCF to use AzureServiceCredentials and set the
+        // StorageSharedKeyCredential instance.
+        serviceBuilder.UseAzureCredentials<Service>(credentials =>
+        {
+            credentials.StorageSharedKey = GetStorageSharedKey();
+        });
+    });
+}
+
+public StorageSharedKeyCredential GetStorageSharedKey()
+{
+    // Fetch shared key using a secure mechanism such as Azure Key Vault
+    // and construct an instance of StorageSharedKeyCredential to return;
+    return default;
+}
+```
+
+Other values for ClientCredentialType are `Sas`, `Token`, and `ConnectionString`. The credentials class `AzureServiceCredentials` has corresponding properties to set each of these credential types.
+
+## Troubleshooting
+
+If there are any errors receiving a message from Azure Queue Storage, the CoreWCF transport will log the details at the `Debug` log level. You can configure logging for the category `Microsoft.CoreWCF` at the `Debug` level to see any errors.
+
+```C# Snippet:CoreWCF_Azure_Storage_Queus_Sample_Logging
+.ConfigureLogging((logging) =>
+{
+    logging.AddFilter("Microsoft.CoreWCF", LogLevel.Debug);
+});
+```
+
+If a message was recevied from the queue, but wasn't able to be processed, it will be placed in the dead letter queue. You can sepcify the dead letter queue name by passing it to the constructor of `AazureQueueStorageBinding`. If not specified, the default value of `default-dead-letter-queue` will be used.
+
+## Key concepts
+
+CoreWCF is an implementation of the service side features of Windows Communication Foundation (WCF) for .NET. The goal of this project is to enable migrating existing WCF services to .NET that are currently using MSMQ and wish to deploy their service to Azure, replacing MSMQ with Azure Queue Storage.
+
+More general samples of using CoreWCF can be found in the [CoreWCF samples repository][corewcf_repo]. To create a client to send messages to an Azure Storage Queue, see the To create a service to receive messages sent to an Azure Storage Queue, see the [Microsoft.WCF.Azure.StorageQueues documentation][wcf_docs]. 
```

Comment:
The second sentence is incomplete. What comes after "see the"?

Reply:
Oops, I'll fix that.

Reply:
I had copied the test from WCF docs and edited it, but I must have pasted it twice. When using clipboard history and using Win-v to paste, if you double click an entry it pastes it twice which is what I think happened here. Anyway, fixed now.